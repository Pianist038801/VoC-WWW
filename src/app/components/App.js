import React from 'react';
import NavigationBar from './NavigationBar';
import SidebarMenu from './SidebarMenu';
import FlashMessagesList from './flash/FlashMessagesList';
//import '../styles/style.css';

class App extends React.Component {
  render() {
    return (
      <div>
        <NavigationBar />
        <FlashMessagesList />
        <SidebarMenu />
        {this.props.children}
      </div>
    );
  }
}

export default App;
