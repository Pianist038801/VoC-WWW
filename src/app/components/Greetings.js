import React from "react";

class Greetings extends React.Component {
  render() {
    return (
      <div className="">
        <h1>Agenta</h1>
      </div>
    );
  }
}
export default Greetings;
