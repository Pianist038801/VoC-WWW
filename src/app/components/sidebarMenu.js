import React from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';

export class SidebarMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: 1
    };
  }
  setSelected(val) {
    this.setState({ selected: val });
  }

  render() {
    const { isAuthenticated } = this.props.auth;

    return (
      <div>
        {isAuthenticated && (
          <section className="sidebar">
            <a href="#">
              <h1 className="logo">
                <span>Agenta</span>
              </h1>
            </a>
            <ul className="list-unstyled">
              <li>
                <a href="#">VOICE OF THE CUSTOMER</a>
              </li>
              <li>
                <Link to="campaigns" onClick={this.setSelected.bind(this, 1)} className={this.state.selected == 1 ? 'active' : ''}>
                  CAMPAIGNS
                </Link>
              </li>
              <li>
                <Link to="surveys" onClick={this.setSelected.bind(this, 2)} className={this.state.selected == 2 ? 'active' : ''}>
                  SURVEYS
                </Link>
              </li>
              <li>
                <Link to="#" onClick={this.setSelected.bind(this, 3)} className={this.state.selected == 3 ? 'active' : ''}>
                  AUDIENCE
                </Link>
              </li>
              <li>
                <Link to="#" onClick={this.setSelected.bind(this, 4)} className={this.state.selected == 4 ? 'active' : ''}>
                  QUESTIONS
                </Link>
              </li>
              <li>
                <Link to="#" onClick={this.setSelected.bind(this, 5)} className={this.state.selected == 5 ? 'active' : ''}>
                  REPORTING
                </Link>
              </li>
              <li>
                <Link to="blacklist" onClick={this.setSelected.bind(this, 6)} className={this.state.selected == 6 ? 'active' : ''}>
                  DO NOT CALL LIST
                </Link>
              </li>
            </ul>
          </section>
        )}
      </div>
    );
  }
}

SidebarMenu.propTypes = {
  auth: React.PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    auth: state.auth
  };
}

export default connect(mapStateToProps, {})(SidebarMenu);
