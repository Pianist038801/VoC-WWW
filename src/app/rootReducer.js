import {combineReducers} from 'redux';
import flashMessages from './reducers/flashMessages';
import auth from './reducers/auth';
import dashboardReducer from './reducers/dashboardReducer';
import global from './reducers/global';
export default combineReducers({flashMessages, auth, dashboardReducer, global});
