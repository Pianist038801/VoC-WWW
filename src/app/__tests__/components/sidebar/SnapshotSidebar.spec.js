import React from 'react';
import { shallow, configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { Provider } from 'react-redux';
import rootReducer from '../../../rootReducer';
import thunk from 'redux-thunk';
import { Router, browserHistory } from 'react-router';
import { createStore, applyMiddleware, compose } from 'redux';
import { SidebarMenu } from '../../../components/SidebarMenu';
import SidebarMenuComponent from '../../../components/SidebarMenu';
import { Link } from 'react-router';
import 'react-datepicker/dist/react-datepicker.css';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import Loader from 'react-loader';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });
const store = createStore(rootReducer, compose(applyMiddleware(thunk), window.devToolsExtension ? window.devToolsExtension() : f => f));

it('matches with the last snapshot', () => {
  let wrapper = shallow(
    <SidebarMenu
      auth={{
        isAuthenticated: true
      }}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
