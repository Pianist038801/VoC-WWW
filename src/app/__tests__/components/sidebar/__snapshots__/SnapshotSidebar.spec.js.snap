// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with the last snapshot 1`] = `
<div>
  <section
    className="sidebar"
  >
    <a
      href="#"
    >
      <h1
        className="logo"
      >
        <span>
          Agenta
        </span>
      </h1>
    </a>
    <ul
      className="list-unstyled"
    >
      <li>
        <a
          href="#"
        >
          VOICE OF THE CUSTOMER
        </a>
      </li>
      <li>
        <Link
          className="active"
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="campaigns"
        >
          CAMPAIGNS
        </Link>
      </li>
      <li>
        <Link
          className=""
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="surveys"
        >
          SURVEYS
        </Link>
      </li>
      <li>
        <Link
          className=""
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="#"
        >
          AUDIENCE
        </Link>
      </li>
      <li>
        <Link
          className=""
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="#"
        >
          QUESTIONS
        </Link>
      </li>
      <li>
        <Link
          className=""
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="#"
        >
          REPORTING
        </Link>
      </li>
      <li>
        <Link
          className=""
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="blacklist"
        >
          DO NOT CALL LIST
        </Link>
      </li>
    </ul>
  </section>
</div>
`;
